src/main/java/org/peergos/util/TraceLogger.java

```diff
@@ -40,13 +45,31 @@ public void setIdentity(PeerId peerId) {
         currentNodeId = nodeIdMap.get(peerId.toString());
     }
 
+    /**
+     * Generates a random Hex string of length traceIdSizeInBytes * 2
+     * 
+     * @return traceId The randomly generated traceId.
+     */
+    public String generateTraceId(int traceIdSizeInBytes) {
```

Comment:
Can this be a private method? Probably moved down the bottom?

Reply:
Yes I think you are right!